### PR TITLE
Add github actions ubuntu-20.04 make build

### PR DESCRIPTION
**Pull request by thomasspriggs**
So that we have testing of the make build on Ubuntu-20.04.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] n/a No code modified - ~Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.~
- [x] n/a No features or user visible behaviour changes - ~The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/~
- [x] n/a No code modified - ~Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).~
- [x] n/a None claimed - ~My commit message includes data points confirming performance improvements (if claimed).~
- [x] My PR is restricted to a single feature or bugfix.
- [x] n/a ~White-space or formatting changes outside the feature-related changed lines are in commits of their own.~

### CODE REVIEW

File: .github/workflows/pull-request-checks.yaml
Comment (hannes-steffenhagen-diffblue) [inline .github/workflows/pull-request-checks.yaml:31]: I believe the unit tests, at least the jbmc ones, actually need to be run from the `unit` directory

Comment (thomasspriggs) [reply .github/workflows/pull-request-checks.yaml:31]: The `-C unit` argument to `make` changes into the `unit` directory and changes back before exiting `make`. This is the same way as the `make` build is implemented in Travis right now.

**Review by hannes-steffenhagen-diffblue — APPROVED** [review]

**Review by hannes-steffenhagen-diffblue — APPROVED** [review]
LGTM with explanation

**Comment by hannes-steffenhagen-diffblue** [comment]
So yeah, the `TESTPL_JOBS` optimisation here leads to a roughly 2x speedup in test runtime. This is expected, but good to have confirmed.

File: .github/workflows/pull-request-checks.yaml
Comment (chrisr-diffblue) [inline .github/workflows/pull-request-checks.yaml:21]: I'd be more inclined to move this `make` invocation into the build section - it's part of the system-under-tests (i.e. the Makefiles) rather than general environment setup (the apt-get)

Comment (NlightNFotis) [reply .github/workflows/pull-request-checks.yaml:21]: Yes, but afaik this doesn't really build minisat, does it? It only downloads the sources and applies a patch as far as I can recall?

Comment (chrisr-diffblue) [reply .github/workflows/pull-request-checks.yaml:21]: All of which is defined in the Makefile, which is what is being tested here.

Comment (NlightNFotis) [reply .github/workflows/pull-request-checks.yaml:21]: Yes, I recall correctly (https://github.com/diffblue/cbmc/blob/develop/src/Makefile#L120).

I would argue this is part of the configuration, as without minisat being downloaded, the build fails - but it's really because the sources are missing, not because of a compilation issue or anything else. And seeing as minisat is one of the SAT solvers we support, I would argue even more this is part of the configuration, even if it's just the default right now.

**Review by NlightNFotis — APPROVED** [review]
LGTM

File: .github/workflows/pull-request-checks.yaml
Comment (chrisr-diffblue) [reply .github/workflows/pull-request-checks.yaml:21]: I don't have a massive issue with this - so I'm happy to approve if everyone else is happy. My logic was:

1) Cmake does the minisat download as part of its configure/build steps, so my suggestion would be more consistent with that
2) If the apt-get fails, or the cpanm fails, its mostly just a case of saying "shrug, infrastructure failed - hit retry", but if the minisat-download fails you _do_ have to double check whether our own stuff needs to be modified. If you have the "setup our CI host" as a separate section to "run our own stuff" then it makes a clear distinction between stuff we may need to manage/modify and stuff that is just pure environment setup.

**Review by chrisr-diffblue — APPROVED** [review]
A minor quibble, but not going to block the PR considering other approvals.

**Pull request merged**
